users/dierkes/preprocessing/segment_cutting.py

```diff
@@ -0,0 +1,107 @@
+import pathlib
+import soundfile as sf
+import numpy as np
+import json
+import multiprocessing
+import argparse
+import tqdm
+import os
+import shutil
+import logging
+
+from sisyphus import Job, Task, tk
+
+class CutSpeechSegmentsFromAudio(Job):
+    """
+    This Job uses a transcription file to cut all segments containing speech from the audio files
+    and stitches them together to files of a target length. Segments that are
+    longer than the target_length are stored as individual files.
+
+
+    """
+    def __init__(self, audio_dir, segments_dir, target_length=20, n_workers=32, file_extension='.wav'):
+        """
+
+        :param audio_dir: path to all the audio files. Subfolders are not yet supported in audio
+            directory
+        :param segments_dir: path to the transcriptions. Subfolders are not yet supported in
+            transcription directory
+        :param target_length: target length of the stitched extracted speech
+        :param n_workers:
+        :param file_extension: currently supported are wav, flac and mp3
+        """
+        self.audio_dir = audio_dir
+        self.segments_dir = segments_dir
+        self.target_length = target_length
+        self.n_workers = n_workers
+        self.file_extension = file_extension
+        assert file_extension in ['.wav', '.flac', '.mp3']
+
+        self.out_audio_folder = self.output_path("audio/", directory=True)
+        self.cut_rqmt = {'time': 8, 'mem': 8, 'cpu': 4}
+
+    def tasks(self):
+        yield Task("cut_speech_segments", rqmt=self.cut_rqmt, mini_task=True)
+
+    def cut_file(self, task):
+        path_file, root_out, target_len_sec, extension, segments = task
+        file_name = path_file.stem
+        meta_file_path = f"{segments}/{file_name}.json"
+
+        with open(meta_file_path, 'r') as f:
+            meta = json.loads(f.read())["value"]["segments"]
+
+        to_cut = []
+        for seg in meta:
+            to_cut += [(seg["start"], seg["end"])]
+
+        data, samplerate = sf.read(path_file)
+        assert len(data.shape) == 1
+        assert samplerate == 8000
```

Comment:
In general, it would be good to have this configurable in the __init__ and then `assert samplerate == self.samplerate`.

Reply:
Or actually not sure if we need to assert this at all. Does it really matter?